pandas/core/dtypes/common.py

```diff
@@ -147,8 +156,18 @@ def is_sparse(arr):
     >>> from scipy.sparse import bsr_matrix
     >>> is_sparse(bsr_matrix([1, 2, 3]))
     False
-    """
 
+    This function checks that 1 dimensional arrays are sparse.
+    It will not identify that a `SparseDataFrame` as sparse.
```

Comment:
We need to update the documentation, but we finally decided to just use "real-world" examples when they help understand the example. In my opinion, in this cases it would be more consistent to use data looking more sparse, for example `SparseSeries([0, 0, 1, 0])`.

Also, I think it'll make things easier for the user if this is written step-by-step. Instead of a longer paragraph explaining all the examples first, having short explanations for each check, for example:
- It returns `True` for if the parameter is a 1D sparse array
- It returns `False` if the parameter is not sparse
- It returns `False` if the parameter has more than 1 dimensions.

It's just an opinion, but personally I think this should be clearer for the user.

Reply:
@datapythonista this is almost an internal function, exposed because we needed a function that says: hey are you a *pandas* sparse array.. we don't have a real sparse type at the moment so this is not a definitive guide. would be ok with improving the doc-string, maybe would should deprecate from the public api.